be/src/exprs/array_functions.cpp

```diff
@@ -1594,4 +1594,137 @@ StatusOr<ColumnPtr> ArrayFunctions::concat(FunctionContext* ctx, const Columns&
     }
 }
 
+template <bool with_length>
+void _array_slice_item(ArrayColumn* column, size_t index, ArrayColumn* dest_column, int64_t offset, int64_t length) {
+    auto& dest_offsets = dest_column->offsets_column()->get_data();
+    if (!offset) {
+        dest_offsets.emplace_back(dest_offsets.back());
+        return;
+    }
+
+    Datum v = column->get(index);
+    const auto& items = v.get<DatumArray>();
+
+    if (offset > 0) {
+        // because offset start with 1.
+        --offset;
+    } else {
+        offset += items.size();
+    }
+
+    auto& dest_data_column = dest_column->elements_column();
+    int64_t end;
+    if constexpr (with_length) {
+        end = std::max((int64_t)0, std::min((int64_t)items.size(), (offset + length)));
+    } else {
+        end = items.size();
+    }
+    offset = (offset > 0 ? offset : 0);
+    for (size_t i = offset; i < end; ++i) {
+        if (items[i].is_null()) {
+            dest_data_column->append_nulls(1);
+        } else {
+            dest_data_column->append_datum(items[i]);
```

Comment:
The `append_datum` interface is not efficient enough...

Reply:
this is the original implementation, I think it's mainly to support complex nested types. BTW, We have no efficient methods for complex nested types